src/storage/src/source/postgres.rs

```diff
@@ -57,78 +57,80 @@ static WAL_LAG_GRACE_PERIOD: Duration = Duration::from_secs(5 * 60); // 5 minute
 static MAX_WAL_LAG: u64 = 100 * 1024 * 1024;
 
 trait ErrorExt {
-    fn is_recoverable(&self) -> bool;
+    fn is_definite(&self) -> bool;
 }
 
 impl ErrorExt for tokio::time::error::Elapsed {
-    fn is_recoverable(&self) -> bool {
-        true
+    fn is_definite(&self) -> bool {
+        false
     }
 }
 
 impl ErrorExt for tokio_postgres::Error {
-    fn is_recoverable(&self) -> bool {
+    fn is_definite(&self) -> bool {
         match self.source() {
-            Some(err) => {
-                match err.downcast_ref::<DbError>() {
-                    Some(db_err) => {
-                        use Severity::*;
-                        // Connection and non-fatal errors
-                        db_err.code() == &SqlState::CONNECTION_EXCEPTION
-                            || db_err.code() == &SqlState::CONNECTION_DOES_NOT_EXIST
-                            || db_err.code() == &SqlState::CONNECTION_FAILURE
-                            || db_err.code() == &SqlState::TOO_MANY_CONNECTIONS
-                            || db_err.code() == &SqlState::CANNOT_CONNECT_NOW
-                            || db_err.code() == &SqlState::ADMIN_SHUTDOWN
-                            || db_err.code() == &SqlState::CRASH_SHUTDOWN
-                            || !matches!(
-                                db_err.parsed_severity(),
-                                Some(Error) | Some(Fatal) | Some(Panic)
-                            )
+            Some(err) => match err.downcast_ref::<DbError>() {
+                Some(db_err) => {
+                    let class = match db_err.code().code().get(0..2) {
+                        None => return false,
+                        Some(class) => class,
+                    };
```

Comment:
til about pg error classes

Reply:
Yeah same TBH! Seemed less error prone than listing out the errors explicitly.